dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/eutaxonomy/AssuranceDataPoint.kt

```diff
@@ -1,18 +1,18 @@
 package org.dataland.datalandbackend.model.eutaxonomy
 
 import com.fasterxml.jackson.annotation.JsonProperty
-import org.dataland.datalandbackend.model.CompanyReportReference
-import org.dataland.datalandbackend.model.enums.eutaxonomy.AssuranceOptions
+import org.dataland.datalandbackend.interfaces.BaseDataPointInterface
+import org.dataland.datalandbackend.model.ExtendedDocumentReference
 
 /**
  * --- API model ---
  * Level of assurance for the reported data and information about the assurance provider
  */
-data class AssuranceData(
+data class AssuranceDataPoint<T>(
```

Comment:
It feels a little overkill using a generic here. As far as I know we always have T = AssuranceOptions. So why not put that in the code directly?

Reply:
Changed